src/backend/distributed/commands/table.c

```diff
@@ -3067,10 +3128,17 @@ ErrorIfUnsupportedAlterTableStmt(AlterTableStmt *alterTableStatement)
 										   "involving partition column")));
 				}
 
+				if (AlterInvolvesIdentityColumn(alterTableStatement, command))
+				{
+					ereport(ERROR, (errmsg("cannot execute ALTER TABLE command "
+										   "involving identity column")));
+				}
+
 				/*
 				 * We check for ALTER COLUMN TYPE ...
 				 * if the column has default coming from a user-defined sequence
-				 * changing the type of the column should not be allowed for now
+				 * or it's an identity column changing the type of the column
```

Comment:
Should move the whole block of comment before the line
`if (AlterInvolvesIdentityColumn(alterTableStatement, command))`